onnxscript/ir/_core_test.py

```diff
@@ -797,7 +797,105 @@ def test_remove_safe_removes_uses_of_removed_nodes(self):
         self.assertEqual(add_node.inputs, (None, None))
 
     # TODO(justinchuby): Test graph mutation methods
+    def test_topological_sort_linear(self):
+        # edges=[(1,2),(2,3)]
+        v0 = _core.Value(name="v0")
+        v1 = _core.Value(name="v1")
+        v2 = _core.Value(name="v2")
+        node1 = _core.Node("", "Node1", inputs=(v0,), num_outputs=1)
+        node2 = _core.Node("", "Node2", inputs=(node1.outputs[0],), num_outputs=1)
+        node3 = _core.Node("", "Node3", inputs=(node2.outputs[0],), num_outputs=1)
+        graph = _core.Graph(
+            (v0, v1, v2),
+            node3.outputs,
+            nodes=(node1, node2, node3),
+            opset_imports={"": 1},
+        )
+        graph.sort()
+        sorted_nodes = tuple(graph)
+        expected_order = (node1, node2, node3)
+        self.assertEqual(sorted_nodes, expected_order)
+
+    def test_topological_sort_independent_subgraphs(self):
+        # edges=[(1,3),(2,4)]
+        v0 = _core.Value(name="v0")
+        v1 = _core.Value(name="v1")
+        v2 = _core.Value(name="v2")
+        v3 = _core.Value(name="v3")
+        node1 = _core.Node("", "Node1", inputs=(v0,), num_outputs=1)
+        node2 = _core.Node("", "Node2", inputs=(v1,), num_outputs=1)
+        node3 = _core.Node("", "Node3", inputs=(node1.outputs[0],), num_outputs=1)
+        node4 = _core.Node("", "Node4", inputs=(node2.outputs[0],), num_outputs=1)
+        graph = _core.Graph(
+            (v0, v1, v2, v3),
+            (node3.outputs[0], node4.outputs[0]),
+            nodes=(node1, node2, node3, node4),
+            opset_imports={"": 1},
+        )
+        graph.sort()
+        sorted_nodes = tuple(graph)
+        # Ensure that dependencies are respected
+        self.assertLess(sorted_nodes.index(node1), sorted_nodes.index(node3))
+        self.assertLess(sorted_nodes.index(node2), sorted_nodes.index(node4))
+        # Ensure all nodes are present
+        self.assertEqual(set(sorted_nodes), {node1, node2, node3, node4})
+
+    def test_topological_sort_shared_dependencies(self):
+        # edges=[(1,3),(2,3)]
+        v0 = _core.Value(name="v0")
+        v1 = _core.Value(name="v1")
+        node1 = _core.Node("", "Node1", inputs=(v0,), num_outputs=1)
+        node2 = _core.Node("", "Node2", inputs=(v0,), num_outputs=1)
+        node3 = _core.Node("", "Node3", inputs=(node1.outputs[0], node2.outputs[0]), num_outputs=1)
+        node4 = _core.Node("", "Node4", inputs=(v1,), num_outputs=1)
+        graph = _core.Graph(
+            (v0, v1),
+            (node3.outputs[0], node4.outputs[0]),
+            nodes=(node1, node2, node3, node4),
+            opset_imports={"": 1},
+        )
+        graph.sort()
+        sorted_nodes = tuple(graph)
+        self.assertLess(sorted_nodes.index(node1), sorted_nodes.index(node3))
+        self.assertLess(sorted_nodes.index(node2), sorted_nodes.index(node3))
+
+    def test_topological_sort_keep_original_1(self):
```

Comment:
Prefer a more descriptive name than `1` and `2`. Consider the following for naming tests:

I would include the expected behavior as part of the test method name. To make what a test is testing clear: `test_<method or function name>_<expected behavior>_[when_<condition>]`.

e.g. `test_method_x_raises_error_when_dims_is_not_a_sequence`